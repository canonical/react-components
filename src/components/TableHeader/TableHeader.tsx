import React, { HTMLProps, ReactNode } from "react";

import { PropsWithSpread, SortDirection } from "types";

export type Props = PropsWithSpread<
  {
    /**
     * The content of the table header.
     */
    children?: ReactNode;
    /**
     * The direction of sorting, if applicable.
     */
    sort?: SortDirection;
  },
  HTMLProps<HTMLTableHeaderCellElement>
>;

const TableHeader = ({ children, sort, ...props }: Props): JSX.Element => {
  return (
    <th role="columnheader" aria-sort={sort} {...props}>
      {children}
    </th>
  );
};

export default TableHeader;
