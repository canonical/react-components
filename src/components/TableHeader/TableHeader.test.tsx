import { render, screen } from "@testing-library/react";
import React from "react";

import TableHeader from "./TableHeader";

describe("TableHeader", () => {
  it("renders", () => {
    render(
      <table>
        <thead>
          <tr>
            <TableHeader>Column 1</TableHeader>
          </tr>
        </thead>
      </table>
    );
    expect(screen.getByRole("columnheader")).toMatchSnapshot();
  });

  it("can set a sort", () => {
    render(
      <table>
        <thead>
          <tr>
            <TableHeader sort="ascending">Column 1</TableHeader>
          </tr>
        </thead>
      </table>
    );
    expect(screen.getByRole("columnheader")).toHaveAttribute(
      "aria-sort",
      "ascending"
    );
  });
});
