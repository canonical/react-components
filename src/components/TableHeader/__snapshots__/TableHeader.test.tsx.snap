// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader renders 1`] = `
<th
  role="columnheader"
>
  Column 1
</th>
`;
